### PR TITLE
Fix install prefix

### PR DESCRIPTION
**Pull request by jwerle**
The `PREFIX` environment variable was not set properly from various `clib` commands. This pull request fixes it.

### CODE REVIEW

File: src/clib-build.c [hunk 1]
```diff
@@ -627,14 +602,37 @@ int main(int argc, char **argv) {
...
   }
 
+  if (!root_package) {
+    const char *name = NULL;
+    char *json = NULL;
+    unsigned int i = 0;
+
+    do {
+      name = manifest_names[i];
+      json = fs_read(name);
```
Comment (stephenmathieson) [inline src/clib-build.c:612]: Doesn't look like we're cleaning up this memory. We should `free(json)` in the `if` below.

File: src/clib-install.c [hunk 2]
```diff
@@ -413,14 +398,37 @@ int main(int argc, char *argv[]) {
...
+    do {
+      name = manifest_names[i];
+      json = fs_read(name);
+    } while (NULL != manifest_names[++i] && !json);
+
+    if (json) {
+      root_package = clib_package_new(json, opts.verbose);
+      if (root_package && root_package->prefix && !opts.prefix) {
+        opts.prefix = root_package->prefix;
+      }
```
Comment (stephenmathieson) [inline src/clib-install.c:415]:
```suggestion
      }
      free(json);
```

File: src/clib-update.c [hunk 3]
```diff
@@ -331,14 +316,37 @@ int main(int argc, char *argv[]) {
...
+    do {
+      name = manifest_names[i];
+      json = fs_read(name);
+    } while (NULL != manifest_names[++i] && !json);
+
+    if (json) {
+      root_package = clib_package_new(json, opts.verbose);
+      if (root_package && root_package->prefix && !opts.prefix) {
+        opts.prefix = root_package->prefix;
+      }
```
Comment (stephenmathieson) [inline src/clib-update.c:333]:
```suggestion
      }
      free(json);
```

**Review by stephenmathieson — CHANGES_REQUESTED** [review]

File: src/clib-build.c (hunk 1 above)
Comment (jwerle) [reply src/clib-build.c:612]: fixed!

**Review by stephenmathieson — APPROVED** [review]
LGTM

Would be nice to have a test for this, but creating one may be more difficult than it's worth.

**Comment by jwerle** [comment]
> LGTM
> 
> Would be nice to have a test for this, but creating one may be more difficult than it's worth.

yeah agreed :]

**Pull request merged**